pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -163,7 +165,7 @@ func (s pausedVMIs) contains(uid types.UID) bool {
 	return ok
 }
 
-func NewLibvirtDomainManager(connection cli.Connection, virtShareDir string, notifier *eventsclient.Notifier, lessPVCSpaceToleration int, agentStore *agentpoller.AsyncAgentStore, ovmfPath string) (DomainManager, error) {
+func NewLibvirtDomainManager(connection cli.Connection, virtShareDir string, notifier *eventsclient.Notifier, lessPVCSpaceToleration int, agentStore *agentpoller.AsyncAgentStore, ovmfPath string, osChosenMigrationProxyPort bool) (DomainManager, error) {
```

Comment:
since this "osChosenMigrationProxyPort"  is an internal testing value, lets keep it entirely isolated (non-exported) within the virtwrap package.

Maybe something like this in manager.go

```
val defaultOSChosenMigrationProxyPort false

func setDefaultOSChosenMigrationProxyPort(val bool) {
defaultOSChosenMigrationProxyPort = val
}

```

then in NewLibvirtDomainManager()

```
osChoseMigrationProxyPort: defaultOSChosenMigrationProxyPort
```

and in manager_test.go

```
BeforeTest() {
setDefaultOSMigrationProxyPort(true)
}
AfterTest() {
setDefaultOSMigrationProxyPort(false)
}
```

Reply:
PR updated, thank you!